src/core/client/admin/components/UserRole/UserRoleChangeContainer.tsx

```diff
@@ -80,22 +137,52 @@ const UserRoleChangeContainer: FunctionComponent<Props> = ({
   return (
     <UserRoleChange
       username={user.username}
-      onChangeRole={handleOnChangeRole}
-      onChangeModerationScopes={handleOnChangeModerationScopes}
-      role={user.role}
+      onChange={onChange}
+      onPromote={onPromote}
+      viewerRole={convertRole(viewer.role)}
+      role={convertRole(user.role)}
       scoped={user.moderationScopes?.scoped}
       moderationScopes={user.moderationScopes}
       moderationScopesEnabled={moderationScopesEnabled}
       query={query}
+      viewerScoped={
+        viewer.moderationScopes ? viewer.moderationScopes?.scoped : false
+      }
+      viewerSites={
+        viewer.moderationScopes
+          ? viewer.moderationScopes.sites?.map((s) => s.id)
+          : null
+      }
```

Comment:
It feels like there's a lot of leakage and duplication of logic in this container component and it's associated control component. Ideally, we should be passing more explicitly what modes to enable for the `UserRoleChange` component rather than giving it all the context to make it's own decisions.